eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java

```diff
@@ -79,27 +85,65 @@ public void exec() throws IOException {
                 }
             }
             ++done;
+            final Map.Entry<Path, Collection<String>> optimized = this.optimize(src);
+            for (final String fqdn : optimized.getValue()) {
+                objects.putIfAbsent(fqdn, new HashSet<>());
+                objects.get(fqdn).add(Save.rel(optimized.getKey()));
+            }
             tojo.set(
                 AssembleMojo.ATTR_XMIR2,
-                this.optimize(src).toAbsolutePath().toString()
+                optimized.getKey().toAbsolutePath().toString()
             );
         }
         if (done > 0) {
             Logger.info(this, "Optimized %d out of %d XMIR program(s)", done, sources.size());
         } else {
             Logger.debug(this, "No XMIR programs out of %d optimized", sources.size());
         }
+        this.checkForDuplicates(objects);
+    }
+
+    /**
+     * Find duplicates in the current compilation scope.
+     * @param objects Top-level compiled objects.
+     */
+    private void checkForDuplicates(final Map<String, Collection<String>> objects) {
+        final Collection<String> duplicates = objects
+            .entrySet()
+            .stream()
+            .filter(entry -> entry.getValue().size() > 1)
+            .map(
+                entry -> {
+                    final String msg = String.format(
+                        "object '%s' defined %d times (%s)",
+                        entry.getKey(),
+                        entry.getValue().size(),
+                        String.join(", ", entry.getValue())
+                    );
+                    Logger.error(this, msg);
```

Comment:
@andreoss I think we should add this "error" to `/program/errors` collection into the XML, instead of logging it here. How those errors are reported later, it's not up to us here. We should do what all other checkers are doing -- register errors in the XML.